frontend/pages/redeem/[urlToken]/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { Box } from '@mui/material';
+import { isString } from 'formik';
+import { NextPage } from 'next';
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import useSWR from 'swr';
+import CampaignCharitySelection from '../../../components/redeem/CampaignCharitySelection';
+import PersonalContribution from '../../../components/redeem/PersonalContribution';
+import RedeemLoading from '../../../components/redeem/RedeemLoading';
+import api from '../../../frontendApis';
+import { CouponRedeemData } from '../../../types/coupons';
+import { SecondaryDonationPostData } from '../../../types/donations';
+import { Nullable } from '../../../types/utils';
+
+const INITIAL_REDEEM_PAGE = 1;
+const NUMBER_OF_REDEEM_PAGES = 2;
+
+const Redeem: NextPage = () => {
+  const router = useRouter();
+  const urlToken = router.query.urlToken && isString(router.query.urlToken) ? router.query.urlToken : null;
+  const { data: coupon, error } = useSWR<Nullable<CouponRedeemData>>([urlToken], (urlToken) =>
+    urlToken !== null ? api.coupons.getCoupon(urlToken).then((r) => r.payload) : null,
```

Comment:
We should have `getUnredeemedCoupon` to only allow people to access unredeemed coupons. Maybe this can be done in another pr but it's a priority.

Reply:
Will be done in another PR. #172